lightweight-term-count-update.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Plugin Name:     Lightweight Term Count Update
+ * Plugin URI:      https://github.com/Automattic/lightweight-term-count-update
+ * Description:     Makes the _update_term_count_on_transition_post_status action very fast
+ * Author:          Automattic
+ * Author URI:      https://automattic.com/
+ * Text Domain:     lightweight-term-count-update
+ * Domain Path:     /languages
+ * Version:         0.1.0
+ *
+ * @package 		Lightweight_Term_Count_Update
+ */
+
+class LTCU_Plugin {
+
+	// post stati which should be counted in term post counting.  Normally at least publish
+	public $counted_stati = array( 'publish' );		
+
+	public function __construct() {
+		add_action( 'init', array( $this, 'init' ), 10, 0 );
+	}
+
+	public function init() {
+		remove_action( 'transition_post_status', '_update_term_count_on_transition_post_status' );
+		add_action( 'transition_post_status', array( $this, 'quick_update_terms_count' ), 10, 3 );
+		$this->counted_stati = apply_filters( 'LTCU_counted_stati',  $this->counted_stati );
+	}
+
+	public function quick_update_terms_count( $new, $old, $post ) {
+		global $wpdb;
+
+		$transition_type = $this->transition_type( $new, $old );
+
+		if ( !$transition_type ) {
+			return false;
+		}
+
+		foreach ( (array) get_object_taxonomies( $post->post_type, 'object' ) as $tax ) {
+			$tt_ids = wp_get_object_terms( $post->ID, $tax->name, array( 'fields' => 'tt_ids' ) );
+			//respect if a taxonomy has a callback override
+			if ( !empty( $tax->update_count_callback ) ) {
+				call_user_func( $tax->update_count_callback, $tt_ids, $tax->name );
```

Comment:
Agree on the type check.  Is the sanitization necessary?  I think it's in core because the tt_ids are passed as arguments whereas we fetch our own immediately before use.  Sanitizing would be fine of course but it's like saying we don't trust the output of `wp_get_object_terms( $post->ID, $tax->name, array( 'fields' => 'tt_ids' ) )` (after we've checked that it's an array.)

Added type check in https://github.com/Automattic/lightweight-term-count-update/pull/1/commits/588d88c98e98c6a56672d712321a812eb2413a28